hermes/src/store.rs

```diff
@@ -277,14 +304,41 @@ impl Store {
             })
             .collect::<Result<Vec<_>>>()?;
 
-        let update_data = construct_update_data(messages.iter().collect())?;
+        let update_data = construct_update_data(messages.into_iter().map(|m| m.into()).collect())?;
 
         Ok(PriceFeedsWithUpdateData {
             price_feeds,
-            wormhole_merkle_update_data: update_data,
+            update_data,
         })
     }
 
+    pub async fn get_price_feeds_with_update_data(
+        &self,
+        price_ids: Vec<PriceIdentifier>,
+        request_time: RequestTime,
+    ) -> Result<PriceFeedsWithUpdateData> {
+        match self
+            .get_price_feeds_with_update_data_from_storage(price_ids.clone(), request_time.clone())
+            .await
+        {
+            Ok(price_feeds_with_update_data) => Ok(price_feeds_with_update_data),
+            Err(e) => {
+                if let RequestTime::FirstAfter(publish_time) = request_time {
+                    if let Some(benchmarks_endpoint) = &self.benchmarks_endpoint {
+                        return benchmarks::get_price_feeds_with_update_data_from_benchmarks(
+                            benchmarks_endpoint.clone(),
+                            price_ids,
+                            publish_time,
+                        )
+                        .await;
+                    }
+                }
+
+                Err(e)
+            }
+        }
```

Comment:
Nitpick: Can use `or_else` to simplify the error handling and remove a level of indentation.

```suggestion
    self.get_price_feeds_with_update_data_from_storage(price_ids.clone(), request_time.clone())
        .await
        .or_else(|e| {
            if let RequestTime::FirstAfter(publish_time) = request_time {
                if let Some(endpoint) = &self.benchmarks_endpoint {
                    return benchmarks::get_price_feeds_with_update_data_from_benchmarks(
                        endpoint.clone(),
                        price_ids,
                        publish_time,
                    )
                    .await;
                }
            }
            Err(e)
        })
```